core/templates/dev/head/services/PlaythroughIssuesService.js

```diff
@@ -127,8 +127,119 @@ oppia.factory('PlaythroughIssuesService', [
+                return index;
+              }
+            },
+            controller: [
+              '$scope', '$uibModalInstance', 'playthroughIndex',
+              'playthrough', 'AlertsService', 'LearnerActionRenderService',
+              function(
+                  $scope, $uibModalInstance, playthroughIndex,
+                  playthrough, AlertsService, LearnerActionRenderService) {
+                $scope.playthroughIndex = playthroughIndex;
+
+                $scope.displayBlocks =
+                  LearnerActionRenderService.getDisplayBlocks(
+                    playthrough.actions);
+                $scope.reversedDisplayBlocks =
+                  $scope.displayBlocks.slice().reverse();
+
+                var blockActionIndexMapping = {};
+                $scope.displayBlocks.reduce(
+                  function(runningTotal, displayBlock, i) {
+                    blockActionIndexMapping[i] =
+                      runningTotal - displayBlock.length;
+                    return blockActionIndexMapping[i];
+                  }, playthrough.actions.length + 1);
+
+                $scope.maxHidden = $scope.displayBlocks.length - 1;
+
+                $scope.getDisplayBlockIndex = function(displayBlock) {
+                  return $scope.displayBlocks.indexOf(displayBlock);
+                };
+
+                $scope.isDisplayBlockOnInitDisplay = function(block) {
+                  return $scope.getDisplayBlockIndex(block) === 0;
+                };
+
+                $scope.createDisplayBlockNavId = function(block) {
+                  return $scope.getDisplayBlockIndex(block) + 1;
+                };
+
+                $scope.renderBlockHtml = function(displayBlock) {
+                  var index = $scope.getDisplayBlockIndex(displayBlock);
+                  return LearnerActionRenderService.renderDisplayBlockHTML(
+                    displayBlock, blockActionIndexMapping[index]);
+                };
+
+                var getRemainingActionsElements = function(pIdx, i) {
+                  return document.getElementById(
+                    'remainingActions' + pIdx.toString() + i.toString());
+                };
+
+                /**
+                 * Shows the remaining display blocks and the arrow div. If
+                 * there is only one display block, the arrow div is not
+                 * shown at all. If the current shown display block is the
+                 * second last display block, the arrow div is hidden after
+                 * the final display block is shown. Else, the following
+                 * display block is displayed.
+                 */
+                $scope.showRemainingActions = function(pIdx) {
+                  // If there is only one display block left to be shown,
+                  // the arrow is not required.
+                  if ($scope.maxHidden === 1) {
+                    getRemainingActionsElements(pIdx, $scope.maxHidden)
```

Comment:
Eww. Shouldn't this be using classes in the HTML? I wouldn't expect to see document.getElementById anywhere in Angular code.

Reply:
Mmm, I think this entire modal could be vastly improved in general, but I'd rather leave that to a separate PR since this one is just deduping the code (and updating this would likely take a lot of test refactoring).

Reply:
OK, could you please file an issue for this explaining exactly what needs to be done? Don't want to lose track of it.

Reply:
Done, I've filed #6433 for this clean-up.